### PR TITLE
Airtable

### PR DESCRIPTION
**Pull request by psavkar**

### CODE REVIEW

File: components/airtable/new-records-in-view.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        },
+        headers: {
+          Authorization: `Bearer ${this.airtable.$auth.api_key}`,
+        },
+      }
+
+      let maxTimestamp = ""
+      const lastMaxTimestamp = this.db.get("lastMaxTimestamp") || ""
+      if (lastMaxTimestamp !== "") {
+        _.set(config,"params.filterByFormula", `CREATED_TIME() > "${lastMaxTimestamp}"`)
```
Comment (tjk) [inline components/airtable/new-records-in-view.js:36]: There is no point to this `_.set`... just use `config.params.filterByFormula` and don't require `lodash`! same in the other file please

File: components/airtable/new-records-in-view.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        intervalSeconds: 60 * 15,
+      },
+    },
+    airtable,
+    baseId: { type: "$.airtable.baseId", appProp: "airtable" },
+    tableId: { type: "$.airtable.tableId", baseIdProp: "baseId" },
+    viewId: { type: "$.airtable.viewId", tableIdProp: "tableId" },
+  },
+  async run(event) {
+      const config = {
```
Comment (tjk) [inline components/airtable/new-records-in-view.js:23]: fix indent?

File: components/airtable/new-records-in-view.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        url: `https://api.airtable.com/v0/${this.baseId}/${encodeURIComponent(this.tableId)}`,
```
Comment (tjk) [inline components/airtable/new-records-in-view.js:24]: why do you encode one interpolation but not the other?

File: components/airtable/new-records-in-view.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      }
+
+      let maxTimestamp = ""
+      const lastMaxTimestamp = this.db.get("lastMaxTimestamp") || ""
+      if (lastMaxTimestamp !== "") {
+        _.set(config,"params.filterByFormula", `CREATED_TIME() > "${lastMaxTimestamp}"`)
+        maxTimestamp = lastMaxTimestamp
+      }
+
+      const response = (await axios(config)).data
```
Comment (tjk) [inline components/airtable/new-records-in-view.js:40]:
```
const { data } = await axios(config)
```

so the variable matches what it is...

File: components/airtable/new-records-in-view.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+      if (response.records.length > 0) {
```
Comment (tjk) [inline components/airtable/new-records-in-view.js:42]: Losing an indent level makes this easier to read imo --

```
if (!response.records.length) {
  console.log(`No new records.`)
  return
}

// rest of the code with one less indent level
```

File: components/airtable/new-records.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    airtable,
+    baseId: { type: "$.airtable.baseId", appProp: "airtable" },
+    tableId: { type: "$.airtable.tableId", baseIdProp: "baseId" },
+  },
+  async run(event) {
+      const config = {
```
Comment (tjk) [inline components/airtable/new-records.js:22]: fix weird indent pls

File: components/airtable/new-records.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        url: `https://api.airtable.com/v0/${this.baseId}/${encodeURIComponent(this.tableId)}`,
```
Comment (tjk) [inline components/airtable/new-records.js:23]: same encodeURIComponent here

File: components/airtable/new-records.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        headers: {
+          Authorization: `Bearer ${this.airtable.$auth.api_key}`,
+        },
+      }
+
+      let maxTimestamp = ""
+      const lastMaxTimestamp = this.db.get("lastMaxTimestamp") || ""
+      if (lastMaxTimestamp !== "") {
+        _.set(config,"params.filterByFormula", `CREATED_TIME() > "${lastMaxTimestamp}"`)
```
Comment (tjk) [inline components/airtable/new-records.js:32]: same

File: components/airtable/new-records.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+  },
+  async run(event) {
+      const config = {
+        url: `https://api.airtable.com/v0/${this.baseId}/${encodeURIComponent(this.tableId)}`,
+        headers: {
+          Authorization: `Bearer ${this.airtable.$auth.api_key}`,
+        },
+      }
+
+      let maxTimestamp = ""
```
Comment (tjk) [inline components/airtable/new-records.js:29]: What is up with these empty string defaults?? are there other falsy values you are guarding against?

```
let maxTimestamp
const lastMaxTimestamp = this.db.get("lastMaxTimestamp")
if (lastMaxTimestamp) {
  config.params.filterByFormula = ...
  maxTimestamp = lastMaxTimestamp
}
```

File: components/airtable/new-records.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+      if (response.records.length > 0) {
+        let recordCount = 0
+        for (let record of response.records) {
+          this.$emit(record, {
+            ts: moment(record.createdTime).valueOf(),
+            summary: JSON.stringify(record.fields),
+            id: record.id,
+          })
+          if (maxTimestamp === '') {
```
Comment (tjk) [inline components/airtable/new-records.js:46]:
```
if (!maxTimestamp || moment(record.createdTime).valueOf() > moment(maxTimestamp).valueOf()) {
  maxTimestamp = record.createdTime
}
```

File: components/airtable/new-records-in-view.js (hunk 2 above)
Comment (psavkar) [reply components/airtable/new-records-in-view.js:23]: done

File: components/airtable/new-records.js (hunk 9 above)
Comment (psavkar) [reply components/airtable/new-records.js:29]: done

File: components/airtable/new-records.js (hunk 7 above)
Comment (psavkar) [reply components/airtable/new-records.js:23]: removed encodeURIComponent

File: components/airtable/new-records.js (hunk 6 above)
Comment (psavkar) [reply components/airtable/new-records.js:22]: done

File: components/airtable/new-records.js (hunk 10 above)
Comment (psavkar) [reply components/airtable/new-records.js:46]: done

File: components/airtable/new-records-in-view.js (hunk 3 above)
Comment (psavkar) [reply components/airtable/new-records-in-view.js:24]: removed

File: components/airtable/new-records-in-view.js (hunk 1 above)
Comment (psavkar) [reply components/airtable/new-records-in-view.js:36]: done

File: components/airtable/new-records-in-view.js (hunk 4 above)
Comment (psavkar) [reply components/airtable/new-records-in-view.js:40]: done

File: components/airtable/new-records-in-view.js (hunk 5 above)
Comment (psavkar) [reply components/airtable/new-records-in-view.js:42]: done

File: components/airtable/new-records.js (hunk 8 above)
Comment (psavkar) [reply components/airtable/new-records.js:32]: done

**Pull request merged**